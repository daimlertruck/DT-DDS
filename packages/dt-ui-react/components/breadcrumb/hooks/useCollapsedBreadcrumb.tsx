import { useTheme } from '@emotion/react';
import {
  Children,
  isValidElement,
  RefObject,
  cloneElement,
  useCallback,
  useRef,
  useState,
  ReactNode,
} from 'react';

import { MoreIcon } from '../../../core';
import useDebounceResize from '../../../hooks/useDebounceResize';
import useMedia from '../../../hooks/useMedia';
import { Breadcrumb } from '../Breadcrumb';

const isBreadcrumbRefValid = (
  breadcrumbRef: RefObject<HTMLElement | null>,
  breadcrumbListRef: RefObject<HTMLElement | null>
) => {
  return !!breadcrumbRef.current && !!breadcrumbListRef.current;
};

export const useCollapsedBreadcrumb = (children: ReactNode) => {
  const breadcrumbRef = useRef<HTMLElement>(null);
  const breadcrumbListRef = useRef<HTMLUListElement>(null);
  const theme = useTheme();
  const isDesktop = useMedia(`(min-width: ${theme.breakpoints.m})`);
  const [visibleChildren, setVisibleChildren] = useState<ReactNode>(children);

  const breadcrumbResize = useCallback(() => {
    if (isDesktop || !isBreadcrumbRefValid(breadcrumbRef, breadcrumbListRef)) {
      setVisibleChildren(children);
      return;
    }

    const childrenArray = Children.toArray(children);
    const isBreadcrumbsTooLarge =
      breadcrumbListRef.current!.scrollWidth >
      breadcrumbRef.current!.clientWidth;

    if (isBreadcrumbsTooLarge) {
      const lastChild = childrenArray[childrenArray.length - 1];
      const modifiedChildren = [
        childrenArray[0],
        <Breadcrumb.Item key='more-icon'>
          <MoreIcon />
        </Breadcrumb.Item>,
        isValidElement(lastChild)
          ? cloneElement(lastChild, {
              ...lastChild.props,
              style: { overflow: 'hidden' },
            })
          : lastChild,
      ];

      setVisibleChildren(modifiedChildren);
    }
  }, [children, isDesktop, breadcrumbRef, breadcrumbListRef]);

  useDebounceResize(breadcrumbResize);

  return { visibleChildren, breadcrumbRef, breadcrumbListRef };
};
