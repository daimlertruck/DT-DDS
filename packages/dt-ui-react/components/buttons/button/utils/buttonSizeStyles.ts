import { CustomTheme as Theme } from '../../../../types';
import { ButtonSizes } from '../constants/types';

export const buttonSizeStyles = (size: ButtonSizes, theme: Theme) =>
  ({
    large: `
      ${theme.fontStyles.button1}
      padding: ${theme.spacing['2xs']} ${theme.spacing.s};
    `,
    medium: `
      ${theme.fontStyles.button2}
      padding: ${theme.spacing['3xs']} ${theme.spacing.xs};
    `,
    small: `
      ${theme.fontStyles.button2}
      padding: ${theme.spacing['4xs']} ${theme.spacing['2xs']};
    `,
  }[size]);
