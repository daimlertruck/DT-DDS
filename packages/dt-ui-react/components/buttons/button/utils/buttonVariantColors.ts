import { CustomTheme as Theme } from '../../../../types';
import {
  ButtonVariant,
  ContainedColors,
  OutlinedColors,
  TextColors,
} from '../constants/types';

interface ColorKeys {
  color: string;
  bgColor: string;
  hoverBgColor?: string;
  hoverColor?: string;
}

export const buttonVariantColors = (
  color: ContainedColors | OutlinedColors,
  theme: Theme
): Record<ButtonVariant, ColorKeys> => ({
  contained: {
    color: theme.colors.alpha.white,
    bgColor: theme.palette.alpha[color],
    hoverBgColor: theme.palette.alpha[`${color}Dark_300` as ContainedColors],
  },
  outlined: {
    color:
      color === 'neutral'
        ? theme.colors.alpha.gray_200
        : theme.palette.alpha[color],
    bgColor: theme.colors.alpha.white,
    hoverBgColor: theme.palette.alpha[`${color}Light_300` as OutlinedColors],
  },
  text: {
    color: theme.palette.alpha[color],
    bgColor: 'transparent',
    hoverColor: theme.palette.alpha[`${color as TextColors}Dark_300`],
  },
});
