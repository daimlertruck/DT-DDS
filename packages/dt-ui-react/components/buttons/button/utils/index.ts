export * from './buttonVariantColors';
export * from './buttonVariantStyles';
