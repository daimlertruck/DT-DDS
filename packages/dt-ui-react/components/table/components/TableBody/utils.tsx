import { Children, ReactNode, cloneElement, isValidElement } from 'react';

import { TableDataCellHeader } from '../TableDataCell/TableDataCell';

export const addHeadersToDataCells = (
  dataCells: ReactNode,
  headers: ReactNode[]
) => {
  return Children.map(dataCells, (dataCell, index) => {
    if (isValidElement(dataCell)) {
      const header = headers[index];
      return cloneElement(
        dataCell,
        {},
        <>
          <TableDataCellHeader>{header}</TableDataCellHeader>
          {dataCell.props.children}
        </>
      );
    }
    return dataCell;
  });
};

export const addHeadersToDataCellsInRows = (
  rows: ReactNode,
  headers: ReactNode[]
) => {
  return Children.map(rows, (row) => {
    if (isValidElement(row)) {
      return cloneElement(
        row,
        {},
        addHeadersToDataCells(row.props.children, headers)
      );
    }
    return row;
  });
};
