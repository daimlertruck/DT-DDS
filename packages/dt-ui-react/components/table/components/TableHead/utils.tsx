import { Children, ReactNode, isValidElement } from 'react';

const extractColumnHeaders = (columnHeaders: ReactNode): ReactNode[] => {
  return Children.toArray(columnHeaders).reduce<ReactNode[]>(
    (headers, columnHeader) => {
      if (isValidElement(columnHeader)) {
        return [...headers, columnHeader.props.children];
      }

      return headers;
    },
    []
  );
};

export const extractTableHeaders = (theadRow: ReactNode) => {
  if (isValidElement(theadRow)) {
    return extractColumnHeaders(theadRow.props.children);
  }

  return [];
};
