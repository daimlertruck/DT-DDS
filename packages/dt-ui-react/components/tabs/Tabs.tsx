import { useMemo, Children, ReactElement, cloneElement } from 'react';

import { BaseProps, Orientation } from '../../types';

import {
  TabsStyled,
  ItemStyled,
  PanelStyled,
  PanelGridStyled,
} from './Tabs.styled';

export interface TabsProps extends BaseProps {
  orientation?: Orientation;
}

type OrientationProps =
  | {
      orientation?: 'vertical';
      hasError?: boolean;
    }
  | {
      orientation?: 'horizontal';
      hasError: false;
    };

export type TabItemProps = {
  activeTab: string;
  index: string;
  handleChange: (value: string) => void;
  isDisabled?: boolean;
} & BaseProps &
  OrientationProps;

export interface TabPanelProps extends BaseProps {
  activeTab: string;
  index: string;
}

export const Tabs = ({
  children,
  dataTestId,
  orientation = 'horizontal',
  style,
}: TabsProps) => {
  const clonedChildren = useMemo(
    () =>
      Children.map(children as ReactElement<TabItemProps>, (child) => {
        return (
          child &&
          cloneElement(child, {
            ...child.props,
            orientation,
          } as TabItemProps)
        );
      }),
    [children, orientation]
  );

  return (
    <TabsStyled
      orientation={orientation}
      data-testid={dataTestId ?? 'tabs'}
      role='tablist'
      style={style}
    >
      {clonedChildren}
    </TabsStyled>
  );
};

Tabs.Item = ({
  activeTab,
  children,
  dataTestId,
  isDisabled,
  hasError,
  index,
  handleChange,
  orientation = 'horizontal',
}: TabItemProps) => {
  return (
    <ItemStyled
      data-testid={dataTestId ?? `tab-item-${index}`}
      active={activeTab === index}
      onClick={() => handleChange(index)}
      disabled={isDisabled}
      role='tab'
      orientation={orientation}
      tabIndex={activeTab === index ? 0 : -1}
      hasError={hasError}
    >
      {children}
    </ItemStyled>
  );
};

export const TabPanel = ({
  style,
  activeTab,
  children,
  dataTestId,
  index,
}: TabPanelProps) => {
  return (
    <PanelStyled
      data-testid={dataTestId ?? `tab-panel-${index}`}
      visible={activeTab === index}
      role='tabpanel'
      tabIndex={activeTab === index ? 0 : -1}
      style={style}
    >
      {children}
    </PanelStyled>
  );
};

TabPanel.Grid = ({ children, dataTestId }: BaseProps) => {
  return (
    <PanelGridStyled data-testid={dataTestId ?? 'tab-panel-grid'}>
      {children}
    </PanelGridStyled>
  );
};
