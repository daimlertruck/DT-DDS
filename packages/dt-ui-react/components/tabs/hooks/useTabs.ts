import { useState, useEffect } from 'react';

export const useTabs = (initialTab: string) => {
  const [activeTab, setActiveTab] = useState<string>(initialTab);
  useEffect(() => {
    setActiveTab(initialTab);
  }, [initialTab]);

  const handleChange = (value: string) => {
    setActiveTab(value);
  };

  return {
    activeTab,
    handleChange,
  };
};
