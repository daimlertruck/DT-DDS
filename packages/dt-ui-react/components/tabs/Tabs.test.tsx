import { render, renderHook, act } from '@testing-library/react';
import React, { Fragment } from 'react';

import { withProviders } from '../../utils';

import { useTabs } from './hooks';
import { Tabs, TabPanel } from './Tabs';

describe('Tabs', () => {
  describe('<Tabs /> component', () => {
    const ProvidedFragment = withProviders(Fragment);

    describe('when orientation is horizontal', () => {
      it('should match the snapshot by displaying the content from Tab 1', () => {
        const handleChange = jest.fn();
        const activeTab = 'first';
        const { container } = render(
          <ProvidedFragment>
            <Tabs>
              <Tabs.Item
                activeTab={activeTab}
                handleChange={handleChange}
                index='first'
              >
                Tab 1
              </Tabs.Item>
              <Tabs.Item
                activeTab={activeTab}
                handleChange={handleChange}
                index='second'
              >
                Tab 2
              </Tabs.Item>
              <Tabs.Item
                activeTab={activeTab}
                handleChange={handleChange}
                index='third'
              >
                Tab 3
              </Tabs.Item>
              <Tabs.Item
                activeTab={activeTab}
                handleChange={handleChange}
                index='fourth'
                isDisabled
              >
                Tab 4
              </Tabs.Item>
            </Tabs>
            <TabPanel activeTab={activeTab} index='first'>
              <TabPanel.Grid>
                <p>Testing tab panel grid:</p>
                <p>Tab 1 content</p>
              </TabPanel.Grid>
            </TabPanel>
            <TabPanel activeTab={activeTab} index='second'>
              Tab 2 content
            </TabPanel>
            <TabPanel activeTab={activeTab} index='third'>
              Tab 3 content
            </TabPanel>
          </ProvidedFragment>
        );

        expect(container).toMatchSnapshot();
      });
    });

    describe('when orientation is vertical', () => {
      it('should match the snapshot by displaying the content from Tab 1', () => {
        const handleChange = jest.fn();
        const activeTab = 'first';
        const { container } = render(
          <ProvidedFragment>
            <Tabs orientation='vertical'>
              <Tabs.Item
                activeTab={activeTab}
                handleChange={handleChange}
                index='first'
              >
                Tab 1
              </Tabs.Item>
              <Tabs.Item
                activeTab={activeTab}
                handleChange={handleChange}
                index='second'
              >
                Tab 2
              </Tabs.Item>
              <Tabs.Item
                activeTab={activeTab}
                handleChange={handleChange}
                index='third'
              >
                Tab 3
              </Tabs.Item>
              <Tabs.Item
                activeTab={activeTab}
                handleChange={handleChange}
                index='fourth'
                isDisabled
              >
                Tab 4
              </Tabs.Item>
            </Tabs>
            <TabPanel activeTab={activeTab} index='first'>
              <TabPanel.Grid>
                <p>Testing tab panel grid:</p>
                <p>Tab 1 content</p>
              </TabPanel.Grid>
            </TabPanel>
            <TabPanel activeTab={activeTab} index='second'>
              Tab 2 content
            </TabPanel>
            <TabPanel activeTab={activeTab} index='third'>
              Tab 3 content
            </TabPanel>
          </ProvidedFragment>
        );

        expect(container).toMatchSnapshot();
      });

      it('should match the snapshot by displaying the red tab', () => {
        const handleChange = jest.fn();
        const activeTab = 'first';
        const { container } = render(
          <ProvidedFragment>
            <Tabs orientation='vertical'>
              <Tabs.Item
                activeTab={activeTab}
                handleChange={handleChange}
                hasError
                index='first'
              >
                Tab 1
              </Tabs.Item>
            </Tabs>
            <TabPanel activeTab={activeTab} index='first'>
              <TabPanel.Grid>
                <p>Testing tab panel grid:</p>
                <p>Tab 1 content</p>
              </TabPanel.Grid>
            </TabPanel>
          </ProvidedFragment>
        );

        expect(container).toMatchSnapshot();
      });
    });
  });

  describe('useTabs hook', () => {
    let hook;
    const initialTab = 'first';
    beforeEach(() => {
      hook = renderHook(() => useTabs(initialTab));
    });

    it('should change active tab', () => {
      act(() => {
        hook.result.current.handleChange('second');

        expect(hook.result.current.activeTab).toEqual(initialTab);
      });
    });
  });
});
