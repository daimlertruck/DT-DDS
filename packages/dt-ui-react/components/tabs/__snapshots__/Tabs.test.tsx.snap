// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs <Tabs /> component when orientation is horizontal should match the snapshot by displaying the content from Tab 1 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 12px;
  margin-bottom: 12px;
}

.emotion-2 {
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: default;
  background-color: #FFFFFF;
  color: #0078D6;
  border-radius: 3px;
  padding: 16px 32px;
}

.emotion-4 {
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  background-color: #E5E5E5;
  color: #4F4F4F;
  border-radius: 3px;
  padding: 16px 32px;
}

.emotion-8 {
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: default;
  pointer-events: none;
  opacity: 0.4;
  background-color: #E5E5E5;
  color: #4F4F4F;
  border-radius: 3px;
  padding: 16px 32px;
}

.emotion-10 {
  background-color: #FFFFFF;
  padding: 32px;
  width: 100%;
}

.emotion-12 {
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-columns: 190px auto;
  gap: 6px;
  overflow: hidden;
  overflow-wrap: break-word;
}

.emotion-14 {
  background-color: #FFFFFF;
  padding: 32px;
  width: 100%;
  display: none;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-testid="tabs"
    orientation="horizontal"
    role="tablist"
  >
    <button
      class="emotion-2 emotion-3"
      data-testid="tab-item-first"
      orientation="horizontal"
      role="tab"
      tabindex="0"
    >
      Tab 1
    </button>
    <button
      class="emotion-4 emotion-3"
      data-testid="tab-item-second"
      orientation="horizontal"
      role="tab"
      tabindex="-1"
    >
      Tab 2
    </button>
    <button
      class="emotion-4 emotion-3"
      data-testid="tab-item-third"
      orientation="horizontal"
      role="tab"
      tabindex="-1"
    >
      Tab 3
    </button>
    <button
      class="emotion-8 emotion-3"
      data-testid="tab-item-fourth"
      disabled=""
      orientation="horizontal"
      role="tab"
      tabindex="-1"
    >
      Tab 4
    </button>
  </div>
  <div
    class="emotion-10 emotion-11"
    data-testid="tab-panel-first"
    role="tabpanel"
    tabindex="0"
  >
    <div
      class="emotion-12 emotion-13"
      data-testid="tab-panel-grid"
    >
      <p>
        Testing tab panel grid:
      </p>
      <p>
        Tab 1 content
      </p>
    </div>
  </div>
  <div
    class="emotion-14 emotion-11"
    data-testid="tab-panel-second"
    role="tabpanel"
    tabindex="-1"
  >
    Tab 2 content
  </div>
  <div
    class="emotion-14 emotion-11"
    data-testid="tab-panel-third"
    role="tabpanel"
    tabindex="-1"
  >
    Tab 3 content
  </div>
</div>
`;

exports[`Tabs <Tabs /> component when orientation is vertical should match the snapshot by displaying the content from Tab 1 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 10px;
}

.emotion-2 {
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: default;
  padding: 16px 14px;
  background-color: #FFFFFF;
  box-shadow: inset 2px 0px 0px 0px #0078D6;
  color: #0078D6;
  max-width: 300px;
  overflow: hidden;
  text-align: start;
}

.emotion-4 {
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding: 16px 14px;
  background-color: #FFFFFF;
  box-shadow: 0;
  color: #8B8C8C;
  max-width: 300px;
  overflow: hidden;
  text-align: start;
}

.emotion-4:hover {
  background: #F4F4F4;
}

.emotion-8 {
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: default;
  pointer-events: none;
  opacity: 0.4;
  padding: 16px 14px;
  background-color: #FFFFFF;
  box-shadow: 0;
  color: #8B8C8C;
  max-width: 300px;
  overflow: hidden;
  text-align: start;
}

.emotion-8:hover {
  background: #F4F4F4;
}

.emotion-10 {
  background-color: #FFFFFF;
  padding: 32px;
  width: 100%;
}

.emotion-12 {
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-columns: 190px auto;
  gap: 6px;
  overflow: hidden;
  overflow-wrap: break-word;
}

.emotion-14 {
  background-color: #FFFFFF;
  padding: 32px;
  width: 100%;
  display: none;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-testid="tabs"
    orientation="vertical"
    role="tablist"
  >
    <button
      class="emotion-2 emotion-3"
      data-testid="tab-item-first"
      orientation="vertical"
      role="tab"
      tabindex="0"
    >
      Tab 1
    </button>
    <button
      class="emotion-4 emotion-3"
      data-testid="tab-item-second"
      orientation="vertical"
      role="tab"
      tabindex="-1"
    >
      Tab 2
    </button>
    <button
      class="emotion-4 emotion-3"
      data-testid="tab-item-third"
      orientation="vertical"
      role="tab"
      tabindex="-1"
    >
      Tab 3
    </button>
    <button
      class="emotion-8 emotion-3"
      data-testid="tab-item-fourth"
      disabled=""
      orientation="vertical"
      role="tab"
      tabindex="-1"
    >
      Tab 4
    </button>
  </div>
  <div
    class="emotion-10 emotion-11"
    data-testid="tab-panel-first"
    role="tabpanel"
    tabindex="0"
  >
    <div
      class="emotion-12 emotion-13"
      data-testid="tab-panel-grid"
    >
      <p>
        Testing tab panel grid:
      </p>
      <p>
        Tab 1 content
      </p>
    </div>
  </div>
  <div
    class="emotion-14 emotion-11"
    data-testid="tab-panel-second"
    role="tabpanel"
    tabindex="-1"
  >
    Tab 2 content
  </div>
  <div
    class="emotion-14 emotion-11"
    data-testid="tab-panel-third"
    role="tabpanel"
    tabindex="-1"
  >
    Tab 3 content
  </div>
</div>
`;

exports[`Tabs <Tabs /> component when orientation is vertical should match the snapshot by displaying the red tab 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 10px;
}

.emotion-2 {
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: default;
  padding: 16px 14px;
  background-color: #FFFFFF;
  box-shadow: inset 2px 0px 0px 0px #D21C1C;
  color: #D21C1C;
  max-width: 300px;
  overflow: hidden;
  text-align: start;
}

.emotion-4 {
  background-color: #FFFFFF;
  padding: 32px;
  width: 100%;
}

.emotion-6 {
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-columns: 190px auto;
  gap: 6px;
  overflow: hidden;
  overflow-wrap: break-word;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-testid="tabs"
    orientation="vertical"
    role="tablist"
  >
    <button
      class="emotion-2 emotion-3"
      data-testid="tab-item-first"
      orientation="vertical"
      role="tab"
      tabindex="0"
    >
      Tab 1
    </button>
  </div>
  <div
    class="emotion-4 emotion-5"
    data-testid="tab-panel-first"
    role="tabpanel"
    tabindex="0"
  >
    <div
      class="emotion-6 emotion-7"
      data-testid="tab-panel-grid"
    >
      <p>
        Testing tab panel grid:
      </p>
      <p>
        Tab 1 content
      </p>
    </div>
  </div>
</div>
`;
